### PR TITLE
MCR-2171 iiif api should be served from api endpoint /api

### PR DESCRIPTION
**Pull request by sebhofmann**
[Link to jira](https://mycore.atlassian.net/browse/MCR-2171).

### CODE REVIEW

**Comment by adlerre** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- mycore-iiif/src/main/java/org/mycore/iiif/common/MCRIIIFBaseURLFilter.java  3
- mycore-iview2/src/main/java/org/mycore/iview2/services/MCRImageThumbnailGenerator.java  7
- mycore-iiif/src/main/java/org/mycore/iiif/common/MCRIIIFApp.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=5511833&bid=18063406)

**Comment by MyCoRe-Travis** [comment]
Something gone wrong on cdebb0b5061ba556ada3100bac8b6a7e64842db3! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/5526-pull_request-966

**Comment by adlerre** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- mycore-iiif/src/main/java/org/mycore/iiif/common/MCRIIIFApp.java  1
- mycore-iview2/src/main/java/org/mycore/iview2/services/MCRImageThumbnailGenerator.java  7
- mycore-iiif/src/main/java/org/mycore/iiif/common/MCRIIIFBaseURLFilter.java  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=5511833&bid=18063406)

File: mycore-iiif/pom.xml [hunk 1]
```diff
@@ -29,5 +33,9 @@
       <groupId>org.mycore</groupId>
       <artifactId>mycore-base</artifactId>
     </dependency>
+    <dependency>
+      <groupId>org.mycore</groupId>
+      <artifactId>mycore-restapi</artifactId>
+    </dependency>
```
Comment (rsteph-de) [inline mycore-iiif/pom.xml:39]: Should we avoid this dependency? - We should keep the components as separate as possible.
Is it an option to move  org.mycore.restapi.MCRJerseyRestApp (with a different name/package) to mycore-base?
There must be some code dealing with Jersey-Resources.

**Review by eckstueck — APPROVED** [review]

File: mycore-iiif/src/main/java/org/mycore/iiif/image/resources/MCRIIIFImageResource.java [hunk 2]
```diff
@@ -114,6 +120,8 @@ public Response getInfoRedirect(@PathParam(IMPL_PARAM) String impl,
 
     @GET
     @Path("{" + IDENTIFIER_PARAM + "}/{region}/{size}/{rotation}/{quality}.{format}")
+    @MCRCacheControl(maxAge = @MCRCacheControl.Age(time = 1, unit = TimeUnit.HOURS),
+        sMaxAge = @MCRCacheControl.Age(time = 1, unit = TimeUnit.HOURS))
```
Comment (yagee-de) [inline mycore-iiif/src/main/java/org/mycore/iiif/image/resources/MCRIIIFImageResource.java]: How often do images change these days? I would prefer a cache time from at least a week.

**Review by yagee-de — CHANGES_REQUESTED** [review]

File: mycore-iiif/src/main/java/org/mycore/iiif/image/resources/MCRIIIFImageResource.java (hunk 2 above)
Comment (sebhofmann) [reply mycore-iiif/src/main/java/org/mycore/iiif/image/resources/MCRIIIFImageResource.java]: A file in a derivate in theory can change all the time, so it basically should never be cached with a age. 
If we set the time to a week or longer, then we limit the use cases for the Interface for a small amount of saved bandwidth.

**Comment by adlerre** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- mycore-iiif/src/main/java/org/mycore/iiif/common/MCRIIIFApp.java  1
- mycore-iview2/src/main/java/org/mycore/iview2/services/MCRImageThumbnailGenerator.java  7
- mycore-iiif/src/main/java/org/mycore/iiif/common/MCRIIIFBaseURLFilter.java  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=5511833&bid=18063406)

**Pull request merged**